### PR TITLE
Restore /potd

### PR DESCRIPTION
**Pull request by mia-pi-git**
It's in `admin.ts` because i was unsure of where else to put it, and admin already has the ProcessManager import. (plus it's console only so...)

### CODE REVIEW

File: server/chat-commands/admin.ts [hunk 1]
```diff
@@ -104,6 +104,25 @@ async function rebuild(context: CommandContext) {
...
+			return this.errorReply(`That Pokemon has no learnset and cannot be used as the PotD.`);
+		}
+		// this _could_ be an instanceof check but worst case we send it to another PM
+		// that's in that file and it does nothing
+		const simProcesses = ProcessManager.processManagers.filter(p => p.basename.includes('room-battle'));
+		for (const sim of simProcesses) {
+			for (const sub of sim.processes) {
+				sub.getProcess().send(`EVAL\n\nConfig.potd = '${species.id}'`);
+			}
+		}
```
Comment (Zarel) [inline server/chat-commands/admin.ts:122]:
```suggestion
		}
		Config.potd = species.id;
```

Comment (Zarel) [inline server/chat-commands/admin.ts:122]: Why are you manually searching through PMs? Isn't this just `Rooms.PM`?

File: server/chat-commands/admin.ts [hunk 2]
```diff
@@ -104,6 +104,25 @@ async function rebuild(context: CommandContext) {
 
 
 export const commands: ChatCommands = {
+	potd(target, room, user) {
+		this.canUseConsole();
+		target = toID(target);
```
Comment (Zarel) [inline server/chat-commands/admin.ts:109]: What's this for?

File: server/chat-commands/admin.ts [hunk 3]
```diff
@@ -104,6 +104,24 @@ async function rebuild(context: CommandContext) {
...
+		if (species.id === Config.potd) {
+			return this.errorReply(`The PotD is already set to ${species.name}`);
+		}
+		if (!species.exists) return this.errorReply(`Pokemon ${target} not found.`);
+		if (!Dex.getLearnsetData(species.id).learnset) {
+			return this.errorReply(`That Pokemon has no learnset and cannot be used as the PotD.`);
+		}
+		Config.potd = species.id;
+		for (const sim of Rooms.PM.processes) {
+			sim.getProcess().send(`EVAL\n\nConfig.potd = '${species.id}'`);
```
Comment (Zarel) [inline server/chat-commands/admin.ts:120]:
```suggestion
		for (const process of Rooms.PM.processes) {
			process.getProcess().send(`EVAL\n\nConfig.potd = '${species.id}'`);
```

File: server/chat-commands/admin.ts (hunk 2 above)
Comment (Zarel) [reply server/chat-commands/admin.ts:109]: `getSpecies` already has an ID conversion, and the only other place `target` is used is an error message that doesn't need to be in ID form.

```suggestion
```

File: server/chat-commands/admin.ts (hunk 1 above)
Comment (mia-pi-git) [reply server/chat-commands/admin.ts]: honestly tho why is this `Rooms.PM` - why not `Rooms.RoomBattle.PM`? It's confusing.

File: server/chat-commands/admin.ts [hunk 4]
```diff
@@ -104,6 +104,23 @@ async function rebuild(context: CommandContext) {
 
 
 export const commands: ChatCommands = {
+	potd(target, room, user) {
+		this.canUseConsole();
+		const species = Dex.getSpecies(target);
+		if (species.id === Config.potd) {
+			return this.errorReply(`The PotD is already set to ${species.name}`);
+		}
+		if (!species.exists) return this.errorReply(`Pokemon ${target} not found.`);
```
Comment (Zarel) [inline server/chat-commands/admin.ts:113]:
```suggestion
		if (!species.exists) return this.errorReply(`Pokemon "${target}" not found.`);
```

**Pull request merged**